### DISCUSSION TITLE
How to switch keyboard layout with hyprctl _fast_?

### DISCUSSION THREAD

**Discussion by MahouShoujoMivutilde** [hyprwm/Hyprland#915] [Q&A] [answered]
#### Usecase

I wanted to make an autocommand for my nvim config to switch to `us` keyboard layout when exiting insert mode (for nvim's bindings to work properly), and switch back to whatever layout I was using when entering insert mode again.

(I am aware of `vim.o.langmap`, it's not perfect, and it's much simpler to just switch layouts).

#### Hyprctl 

So far I see that I can switch layouts by basically _redefining_ them with:

```sh
hyprctl keyword input:kb_layout us
```
You can even pass multiple layouts in different order to switch to the needed (first), while keeping the ability to toggle between them without `hyprctl`.

But the problem is - it's kind of slow:

```
❯ hyperfine "hyprctl keyword input:kb_layout us"
Benchmark 1: hyprctl keyword input:kb_layout us
  Time (mean ± σ):      41.5 ms ±   6.6 ms    [User: 0.6 ms, System: 0.7 ms]
  Range (min … max):    35.0 ms …  73.4 ms    38 runs
 
```

But the actual proper layout switching (without redefining them, like above, the one like when you press right alt when `kb_options = grp:toggle` is defined) - it's pretty fast, as far as I can tell, I don't feel any input lag with it.

#### Question

So, basically, my question is this:

Is there anything like sway's 

```
swaymsg input type:keyboard xkb_switch_layout <next|prev|index>
```
?

From the man page:
> `input <identifier> xkb_switch_layout <index>|next|prev`
> Changes the active keyboard layout to <index> counting from zero or to next or previous layout on the list. If there is no next or previous layout, this command hops to the other end of the list.
> This can be used when multiple layouts are configured with xkb_layout. A list of layouts you can switch between can be obtained with swaymsg -t get_inputs.

https://man.archlinux.org/man/sway-input.5.en

**Comment by vaxerski** [comment]
nope. //2short

If you need absolute speed try using the xkb `grp:`toggles

**Reply by MahouShoujoMivutilde** [reply]
\*sad modal editor noises\*

**Reply by MahouShoujoMivutilde** [reply]
I use one already to switch layouts manually, but how do I trigger it with `hyprctl`?

**Reply by vaxerski** [reply]
u cant

**Comment by vaxerski** [comment, marked as answer]
reposting for whoever will read this in the future, see 79adba557eccff5d19f300f3d04b1f3f67aee407 and #1163
